diff --git a/way2sms.py b/way2sms.py
index b3bd23f..18f3688 100644
--- a/way2sms.py
+++ b/way2sms.py
@@ -65,8 +65,8 @@ class sms:
 		'''
 		Sends the message to the given mobile number
 		'''
-
-		if len(msg)>139 or len(mobile_no)!=10 or not mobile_no.isdecimal():	#checks whether the given message is of length more than 139
+                mobile_no_unicode = mobile_no.decode('utf-8')
+		if len(msg)>139 or len(mobile_no)!=10 or not mobile_no_unicode.isdecimal():	#checks whether the given message is of length more than 139
 
 			return False							#or the mobile_no is valid
 
@@ -91,7 +91,8 @@ class sms:
 	def sendLater(self, mobile_no, msg, date, time):				#Function for future SMS feature.
 											#date must be in dd/mm/yyyy format
 											#time must be in 24hr format. For ex: 18:05
-		if len(msg)>139 or len(mobile_no)!=10 or not mobile_no.isdecimal():
+                mobile_no_unicode = mobile_no.decode('utf-8')
+		if len(msg)>139 or len(mobile_no)!=10 or not mobile_no_unicode.isdecimal():
 			return False
 
 		dateparts = date.split('/')			#These steps to check for valid date and time and formatting
